### ISSUE TITLE
Node classification with Node2Vec "Data Splitting" section inconsistent

### ISSUE THREAD

**Issue by ankitatalwar** [stellargraph/stellargraph#1778]
### Describe the bug

The "Data Splitting" section in Node Classification with Node2Vec claims to make a 75-25 split, however the code makes a 90-10 split. Also, the code has wrongly assigned train and test sets. It assigns 10% data to the train set and 90% to the test set.

### To Reproduce

https://stellargraph.readthedocs.io/en/stable/demos/node-classification/node2vec-node-classification.html#Data-Splitting

### Observed behavior

We split the data into train and test sets.
We use 75% of the data for training and the remaining 25% for testing as a hold out test set.

X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.1, test_size=None)

### Expected behavior

X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.75, test_size=None)

### Environment
https://stellargraph.readthedocs.io/en/stable/demos/node-classification/node2vec-node-classification.html#Data-Splitting

**Comment by huonw** [comment]
Thanks for filing an issue! You're entirely correct that the text and code are inconsistent here.

It looks like this was correct and consistent in the original form of the notebook (added in 7d14eb28ab3254aae2c8c64e70a9fc2357bb83af) https://github.com/stellargraph/stellargraph/blob/7d14eb28ab3254aae2c8c64e70a9fc2357bb83af/demos/node-classification/stellargraph-node2vec-node-classification.ipynb but was changed in https://github.com/stellargraph/stellargraph/commit/7a6742213e0e8765bc6b712f9783a13a80199abb with apparently no explanation.

There's a few options:

1. update the text to match the code (that is, "We use 10% of the data for training and the remaining 90% for testing as a hold out test set"). For example: https://stellargraph.readthedocs.io/en/stable/demos/node-classification/keras-node2vec-node-classification.html#Data-Splitting does 10-90 'properly'
2. update the code to match the text (that is, what you state). For example: https://stellargraph.readthedocs.io/en/stable/demos/node-classification/node2vec-weighted-node-classification.html#Comparing-the-accuracy-of-node-classification-for-weighted-(weight-==1)-and-unweighted-random-walks.
3. switch both to something completely different. For example: https://stellargraph.readthedocs.io/en/stable/demos/node-classification/attri2vec-node-classification.html#Data-Splitting does a 20-80 split

I'm inclined towards 2; potentially including updating those other examples that do splits other than 75-25 because I suspect at least the notebook from one steals the 10-90 split from the new weird notebook.

I'm happy to open a PR for this sometime later today or tomorrow, unless someone else gets to it first.
